app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/VideoProjectionMenuWidget.java

```diff
@@ -116,4 +117,37 @@ private void handleClick(@VideoProjectionFlags int aVideoProjection) {
     public @VideoProjectionFlags int getSelectedProjection() {
         return mSelectedProjection;
     }
+
+    public static int getAutomaticProjection(String aURL) {
+        if (aURL == null) {
+            return -1;
+        }
+
+        Uri uri = Uri.parse(aURL);
+        if (uri == null) {
+            return -1;
+        }
+
+        String projection = uri.getQueryParameter("mozVideoProjection");
+        if (projection == null) {
+            return -1;
+        }
+        projection = projection.toLowerCase();
+
+        if (projection.equals("360")) {
+            return VIDEO_PROJECTION_360;
+        } else if (projection.equals("360s")) {
+            return VIDEO_PROJECTION_360_STEREO;
+        } else if (projection.equals("180")) {
+            return VIDEO_PROJECTION_180;
+        } else if (projection.equals("180lr")) {
+            return VIDEO_PROJECTION_180_STEREO_LEFT_RIGTH;
+        } else if (projection.equals("180tp")) {
+            return VIDEO_PROJECTION_180_STEREO_TOP_BOTTOM;
+        } else if (projection.equals("3d")) {
```

Comment:
what about 3D top bottom?

Reply:
Usually referred to as "over-under". 
Also Left/Right is usually referred to "side-by-side" or _SBS_

We shouldn't refer to video as 3D.  It's stereo 360 or stereo 180, both being typically SBS (mostly assumed).